carousel/core/models.py

```diff
@@ -218,7 +218,16 @@ def _initialize(self):
             if src_value:
                 value = src_model[layer] = src_value
             else:
-                value = dict(value['sources'])
+                srcmod, srcpkg = value.get('module'), value.get('package')
+                try:
+                    value = dict(value['sources'])
+                except ValueError:
+                    value = dict.fromkeys(value['sources'], {})
+                for src in value.viewkeys():
+                    if srcmod is not None:
+                        value[src]['module'] = srcmod
+                    if srcpkg is not None:
+                        value[src]['package'] = srcpkg
             # set layer attribute with model data
             setattr(self, layer, layer_cls(value))
```

Comment:
Ok I think I understand better now, it seems like value ends up looking like (for instance):
```
{ 'IrradianceCalcs' : { 'module': '.....', 'package': '.......'}, ... }
```,
which is then passed to the layer ``__init__`` as the ``sources`` argument, which ends up in the ``layer`` attribute of the layer.